### PR TITLE
samples: loadsw

### PR DESCRIPTION
**Pull request by wilu-nordic**

### CODE REVIEW

File: samples/loadsw/README.rst [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+Configuration options
+=====================
+
+Check and configure the following sample-specific Kconfig options to specify which testcase you want to run:
+
+.. CONFIG_TESTCASE_SW_ENABLE:
+
+CONFIG_TESTCASE_SW_ENABLE
+  This sample configuration enables controlling LOADSWs using software.
```
Comment (annakielar) [inline samples/loadsw/README.rst:82]:
```suggestion
  This sample configuration enables controlling LOADSWs through the software.
```

File: samples/loadsw/README.rst [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  In an infinite loop, switches are enabled and disabled alternately, with a delay of 5 seconds in between.
+
+.. CONFIG_TESTCASE_GPIO_ENABLE:
+
+CONFIG_TESTCASE_GPIO_ENABLE
+  This sample configuration enables controlling LOADSWs using GPIOs.
```
Comment (annakielar) [inline samples/loadsw/README.rst:88]:
```suggestion
  This sample configuration enables controlling LOADSWs through GPIOs.
```

Comment (annakielar) [reply samples/loadsw/README.rst:88]: Comparing to line 24 - GPIO or GPIOs?

File: samples/loadsw/README.rst [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  GPIO1 and GPIO2 are switched to input mode and selected GPIO pins (`NPMX_LDSW_GPIO_1` and `NPMX_LDSW_GPIO2`) are configured to control the switches.
+  When you change the input states of the selected GPIOs you should see voltage changes in the load switches 1 and 2 outputs.
```
Comment (annakielar) [inline samples/loadsw/README.rst:90]:
```suggestion
  GPIO1 and GPIO2 are switched to the input mode, and selected GPIO pins (`NPMX_LDSW_GPIO_1` and `NPMX_LDSW_GPIO2`) are configured to control the switches.
  When you change the input states of the selected GPIOs, you should see voltage changes in both 1 and 2 load switches' outputs.
```

File: samples/loadsw/README.rst [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+   * For advanced output voltage testcase: Check output voltage on EK **VOUT1** pin - it should start from 1.0 V and increase by 200 ms up to 3.3 V.
+   * For retention voltage testcase: Connect EK **GPIO1** pin to GND.
+     Check output voltage on EK **VOUT1** pin - it should be 1.7 V.
+     Connect EK **GPIO1** pin to VDD.
+     Check output voltage on EK **VOUT1** pin - it should be 3.3 V.
+   * For enable LOADSWs using pin testcase:
+     Connect EK **GPIO1** or **GPIO2** pin to GND.
+     Check output voltage on EK **LSOUT1** or **LSOUT2** - it should be 0.0 V.
+     Connect EK **GPIO1** or **GPIO2** pin to VDD.
+     Check output voltage on EK **LSOUT1** or **LSOUT2** pin - it should be 3.3 V or 1.7 V.
```
Comment (annakielar) [inline samples/loadsw/README.rst:124]:
```suggestion
   * For simple output voltage testcase, check output voltage on EK **VOUT1** pin - it should be 2.4 V.
   * For advanced output voltage testcase, check output voltage on EK **VOUT1** pin - it should start from 1.0 V and increase by 200 ms up to 3.3 V.
   * For retention voltage testcase, do the following:
   
     #. Connect EK **GPIO1** pin to GND.
     #. Check output voltage on EK **VOUT1** pin - it should be 1.7 V.
     #. Connect EK **GPIO1** pin to VDD.
     #. Check output voltage on EK **VOUT1** pin - it should be 3.3 V.
     
   * For enable LOADSWs using pin testcase, do the following:
     
     #. Connect EK **GPIO1** or **GPIO2** pin to GND.
     #. Check output voltage on EK **LSOUT1** or **LSOUT2** - it should be 0.0 V.
     #. Connect EK **GPIO1** or **GPIO2** pin to VDD.
     #. Check output voltage on EK **LSOUT1** or **LSOUT2** pin - it should be 3.3 V or 1.7 V.
```

File: samples/loadsw/src/main.c [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+	/* Get the pointer to npmx device. */
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+
+	npmx_buck_t *buck = npmx_buck_get(npmx_instance, 0);
+
+	/* Set the output voltage of BUCK1. */
+	npmx_buck_normal_voltage_set(buck, NPMX_BUCK_VOLTAGE_1V8);
+
+	/* Have to be called each time to change output voltage of the BUCK1. */
```
Comment (annakielar) [inline samples/loadsw/src/main.c:135]:
```suggestion
	/* Has to be called each time to change output voltage of the BUCK1. */
```
BTW what has to be called?

File: samples/loadsw/src/main.c [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+	npmx_buck_vout_select_set(buck, NPMX_BUCK_VOUT_SELECT_SOFTWARE);
+
+	/* Enable BUCK1. */
+	npmx_buck_task_trigger(buck, NPMX_BUCK_TASK_ENABLE);
+
+	npmx_ldsw_t *ldsw_1 = npmx_ldsw_get(npmx_instance, 0);
+	npmx_ldsw_t *ldsw_2 = npmx_ldsw_get(npmx_instance, 1);
+
+	/* Enable active discharge, to see changes on output pins. */
```
Comment (annakielar) [inline samples/loadsw/src/main.c:144]:
```suggestion
	/* Enable active discharge to see changes on output pins. */
```

**Review by annakielar — CHANGES_REQUESTED** [review]

**Review by annakielar — CHANGES_REQUESTED** [review]

**Review by annakielar — CHANGES_REQUESTED** [review]

**Review by annakielar — DISMISSED** [review]

File: samples/ldsw/README.rst [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+=======
+
+|test_sample|
+
+#. |connect_kit|
+#. |connect_terminal|
+#. Connect a multimeter to **LSOUT1** and **LSOUT2**.
+#. For the `CONFIG_TESTCASE_GPIO_ENABLE`_ testcase, do the following:
+
+   a. Connect the **GPIO1** EK pin to GND and  the **GPIO2** EK pin to VDD.
```
Comment (annakielar) [inline samples/ldsw/README.rst:110]:
```suggestion
   a. Connect the **GPIO1** EK pin to GND and the **GPIO2** EK pin to VDD.
```

**Review by annakielar — APPROVED** [review]

File: samples/ldsw/src/main.c [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+
+	/* Set the output voltage of BUCK1. */
+	npmx_buck_normal_voltage_set(buck, NPMX_BUCK_VOLTAGE_1V8);
+	npmx_buck_vout_select_set(buck, NPMX_BUCK_VOUT_SELECT_SOFTWARE);
+
+	/* Enable BUCK1. */
+	npmx_buck_task_trigger(buck, NPMX_BUCK_TASK_ENABLE);
+
+	npmx_ldsw_t *ldsw_1 = npmx_ldsw_get(npmx_instance, 0);
+	npmx_ldsw_t *ldsw_2 = npmx_ldsw_get(npmx_instance, 1);
```
Comment (masz-nordic) [inline samples/ldsw/src/main.c:140]:
```suggestion
	npmx_ldsw_t *ldsw_0 = npmx_ldsw_get(npmx_instance, 0);
	npmx_ldsw_t *ldsw_1 = npmx_ldsw_get(npmx_instance, 1);
```

Comment (wilu-nordic) [reply samples/ldsw/src/main.c:140]: But in IPS numeration is `1` and `2` and we decided to be compatible.

Comment (wilu-nordic) [reply samples/ldsw/src/main.c:140]: Can we postpone this until https://nordicsemi.atlassian.net/browse/NEU-3407 will be solved?

**Review by masz-nordic — APPROVED** [review]

**Pull request merged**